packages/audit/audit-testsuite/runtest.sh

```diff
@@ -0,0 +1,157 @@
+#   Copyright (c) 2010 Red Hat, Inc. All rights reserved.
+#
+#   This copyrighted material is made available to anyone wishing
+#   to use, modify, copy, or redistribute it subject to the terms
+#   and conditions of the GNU General Public License version 2.
+#
+#   This program is distributed in the hope that it will be
+#   useful, but WITHOUT ANY WARRANTY; without even the implied
+#   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#   PURPOSE. See the GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public
+#   License along with this program; if not, write to the Free
+#   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+#   Boston, MA 02110-1301, USA.
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Include rhts environment
+. /usr/bin/rhts-environment.sh || exit 1
+. /usr/lib/beakerlib/beakerlib.sh || exit 1
+
+PACKAGE="kernel"
+
+# Optional test parametr - location of testuite git.
+GIT_URL=${GIT_URL:-"https://github.com/linux-audit/audit-testsuite.git"}
+
+# Optional test paramenter - branch containing tests.
+GIT_BRANCH=${GIT_BRANCH:-"master"}
+
+# Optional test parameter - list to tests to be executed.
+TESTS=${TESTS:-""}
+
+# Optional test parameter - debug mode.
+DEBUG=${DEBUG:-""}
+
+# Testing directory.
+TEST_DIR=$(mktemp -d)
+
+# Architecture.
+ARCH=$(uname -m)
+
+rlJournalStart
+
+    rlPhaseStartSetup
+
+        # Check requirements.
+        rlCheckRpm "kernel"
+        rlCheckRpm "git"
+        rlCheckRpm "audit"
+        rlCheckRpm "perl"
+        rlIsFedora && rlCheckRpm "perl-Test"
+        rlCheckRpm "perl-Test-Harness"
+        rlCheckRpm "perl-File-Which"
+        rlCheckRpm "perl-Time-HiRes"
+        rlCheckRpm "libgcc"
+        rlCheckRpm "expect"
+
+        case $ARCH in
+            "x86_64")
+                rlCheckRpm "glibc-devel.i686"
+                rlCheckRpm "libgcc.i686"
+                ;;
+            "s390x")
+                rlCheckRpm "glibc-devel.s390"
+                rlCheckRpm "libgcc.s390"
+                ;;
+            "ppc64le")
+                rlCheckRpm "glibc-devel.ppc"
+                rlCheckRpm "libgcc.ppc"
+                ;;
+            "ppc64")
+                rlCheckRpm "glibc-devel.ppc"
+                rlCheckRpm "libgcc.ppc"
+                ;;
+        esac
+
+        # Download testsuite.
+        rlRun "git clone $GIT_URL" 0
+        if [ $? != 0 ]; then
+            echo "Failed to git clone $GIT_URL." | tee -a $OUTPUTFILE
+            rhts-report-result $TEST WARN $OUTPUTFILE
+            rhts-abort -t recipe
+        fi
+
+        rlRun "pushd audit-testsuite" 0 || rlDie
+        rlRun "git checkout $GIT_BRANCH" 0 || rlDie
+
+        # Apply workarounds for beaker environment.
+        rlRun "unset DISTRO" 0
+        rlRun "echo $(id -u) >/proc/self/loginuid" 0
+
+	# Turn off intel-only test on non-intel architectures.
+	test "$(rlGetPrimaryArch)" != "x86_64" && \
```

Comment:
I'm a bit confused by this check -- the comment says "intel-only" but x86_64 covers AMD too? Either the comment should be updated or the code is wrong and the test will fail on AMD machines.

Other than this I don't see anything weird in the test, but I'm not familiar with it :)

Reply:
That comment is from  original internal repo source code., just realize the comment is misleading  :-) 

This the full explanation:
When  running on  Non arch:x86_64. the specific test must be removed since only supported on x86_64  architecture only. :-D

I can change  the comment to "# Turn off x86_64 specific test  when running on non x86_64 architectures"

is that ok?

Reply:
Yes that works :) thanks for explanation!

Reply:
@veruu, Thanks :-D

Please re-review again the updated comment.